app/application/services/subject_type_schema_validator.py

```
@@ -0,0 +1,49 @@
+class SubjectTypeSchemaValidator:
+    """Validates subject attributes against subject type schema when type is configured with a schema."""
+
+    def __init__(self, subject_type_repo: "ISubjectTypeRepository") -> None:
+        self._repo = subject_type_repo
+
+    async def validate_attributes(
+        self,
+        tenant_id: str,
+        subject_type: str,
+        attributes: dict[str, Any],
+        display_name: str | None = None,
```

Ruff (0.15.0)

[warning] Unused method argument: `display_name`

(ARG002)